src/main/java/seedu/address/logic/parser/appointmentparser/EditAppointmentCommandParser.java

```diff
@@ -52,21 +58,36 @@ public EditAppointmentCommand parse(String args) throws ParseException {
                     argMultimap.getValue(PREFIX_SUBJECT_NAME).get()));
         }
 
-        if (argMultimap.getValue(PREFIX_DATE).isPresent() && argMultimap.getValue(PREFIX_TIME_FROM).isPresent()) {
+        if (argMultimap.getValue(PREFIX_DATE).isPresent()
+                || argMultimap.getValue(PREFIX_TIME_FROM).isPresent()
+                || argMultimap.getValue(PREFIX_TIME_TO).isPresent()) {
+            if (!ArgumentTokenizer.arePrefixesPresent(argMultimap, PREFIX_TIME_FROM,
+                    PREFIX_TIME_TO)) {
+                throw new ParseException(MESSAGE_MISSING_DATE_FIELD);
+            }
+        }
```

Comment:
Why does this only check for PREFIX_TIME_FROM and PREFIX_TIME_TO? What if PREFIX_TIME_FROM is present but PREFIX DATE is missing?

Reply:
Fixed